core/common/boot/src/main/java/org/eclipse/edc/boot/system/ExtensionLoader.java

```diff
@@ -112,6 +112,21 @@ private static Supplier<Object> getDefaultProviderInvoker(ServiceExtensionContex
         return openTelemetries.isEmpty() ? GlobalOpenTelemetry.get() : openTelemetries.get(0);
     }
 
+    /**
+     * Parses the ConsoleMonitor log level from the program args. If no log level is provided, defaults to Level default.
+     */
+    private static ConsoleMonitor.Level parseLogLevel(String[] programArgs) {
+        return Set.of(programArgs).stream().filter(arg -> arg.startsWith(ConsoleMonitor.LEVEL_PROG_ARG)).map(arg -> getLogLevel(arg.split("=")[1])).findFirst().orElse(ConsoleMonitor.Level.getDefaultLevel());
```

Comment:
nit: it could be formatted to improve readability (and make the comment not necessary):
```java
return Set.of(programArgs).stream()
    .filter(arg -> arg.startsWith(ConsoleMonitor.LEVEL_PROG_ARG))
    .map(arg -> getLogLevel(arg.split("=")[1]))
    .findFirst()
    .orElse(ConsoleMonitor.Level.getDefaultLevel());
```